features/preferences/impl/src/main/kotlin/io/element/android/features/preferences/impl/notifications/NotificationSettingsView.kt

```diff
@@ -0,0 +1,266 @@
+    onBackPressed: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    OnLifecycleEvent { _, event ->
+        when (event) {
+            Lifecycle.Event.ON_RESUME -> state.eventSink.invoke(NotificationSettingsEvents.RefreshSystemNotificationsEnabled)
+            else -> Unit
+        }
+    }
+    PreferenceView(
+        modifier = modifier,
+        onBackPressed = onBackPressed,
+        title = stringResource(id = CommonStrings.screen_notification_settings_title)
+    ) {
+
+        when (state.matrixSettings) {
+            is NotificationSettingsState.MatrixSettings.Invalid -> InvalidNotificationSettingsView(
+                showError = state.matrixSettings.fixFailed,
+                onContinueClicked = { state.eventSink(NotificationSettingsEvents.FixConfigurationMismatch) },
+                onDismissError = { state.eventSink(NotificationSettingsEvents.ClearConfigurationMismatchError) },
+            )
+            NotificationSettingsState.MatrixSettings.Uninitialized -> return@PreferenceView
+            is NotificationSettingsState.MatrixSettings.Valid -> NotificationSettingsContentView(
+                matrixSettings = state.matrixSettings,
+                systemSettings = state.appSettings,
+                onNotificationsEnabledChanged = { state.eventSink(NotificationSettingsEvents.SetNotificationsEnabled(it))},
+                onGroupChatsClicked = { onOpenEditDefault(false) },
+                onDirectChatsClicked = { onOpenEditDefault(true) },
+                onMentionNotificationsChanged = { state.eventSink(NotificationSettingsEvents.SetAtRoomNotificationsEnabled(it)) },
+                onCallsNotificationsChanged = { state.eventSink(NotificationSettingsEvents.SetCallNotificationsEnabled(it)) },
+            )
+        }
+    }
+}
+
+@Composable
+private fun NotificationSettingsContentView(
+    matrixSettings: NotificationSettingsState.MatrixSettings.Valid,
+    systemSettings: NotificationSettingsState.AppSettings,
+    onNotificationsEnabledChanged: (Boolean) -> Unit,
+    onGroupChatsClicked: () -> Unit,
+    onDirectChatsClicked: () -> Unit,
+    onMentionNotificationsChanged: (Boolean) -> Unit,
+    onCallsNotificationsChanged: (Boolean) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+        val context = LocalContext.current
+        if (systemSettings.appNotificationsEnabled && !systemSettings.systemNotificationsEnabled) {
+            PreferenceText(
+                icon = Icons.Filled.NotificationsOff,
+                title = stringResource(id = CommonStrings.screen_notification_settings_system_notifications_turned_off),
+                subtitle = stringResource(id = CommonStrings.screen_notification_settings_system_notifications_action_required,
+                    stringResource(id = CommonStrings.screen_notification_settings_system_notifications_action_required_content_link)),
+                onClick = {
+                    val intent = Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS)
```

Comment:
We have `Context.startNotificationSettingsIntent` in the `androidutils` library module, but it uses an `ActivityResultLauncher`. Maybe we could create a version without that and re-use it here?